jobs/yolov5-web-service/app.py

```diff
@@ -0,0 +1,63 @@
+import io
+import os
+import shutil
+import json
+from PIL import Image
+
+import torch
+import flask
+from flask import Flask, jsonify, url_for, render_template, request, redirect
+
+app = Flask(__name__)
+
+RESULT_FOLDER = os.path.join('static')
+app.config['RESULT_FOLDER'] = RESULT_FOLDER
+
+model_yolov5s = torch.hub.load('ultralytics/yolov5', 'yolov5s', pretrained=True).autoshape() 
+model_yolov5s.eval()
+
+model_yolov5m = torch.hub.load('ultralytics/yolov5', 'yolov5m', pretrained=True).autoshape() 
+model_yolov5m.eval()
+
+# yolov5s
+def get_prediction_yolov5s(img_bytes):
+    img = Image.open(io.BytesIO(img_bytes))
+    imgs = [img]  # batched list of images
+    # Inference
+    results = model_yolov5s(imgs, size=640) 
+    return results
+
+# yolov5m
+def get_prediction_yolov5m(img_bytes):
+    img = Image.open(io.BytesIO(img_bytes))
+    imgs = [img]  # batched list of images
+    # Inference
+    results = model_yolov5m(imgs, size=640) 
+    return results
+
+@app.route('/', methods=['GET', 'POST'])
+def predict():
+    if request.method == 'POST':
+        # model choice
+        print(f'User selected model : {request.form.get("model_choice")}')
+
+        if 'file' not in request.files:
+            return redirect(request.url)
+        file = request.files.get('file')
+        if not file:
+            return
+        
+        img_bytes = file.read()
+        if request.form.get("model_choice") == 'yolov5s':
+            results = get_prediction_yolov5s(img_bytes)
+        if request.form.get("model_choice") == 'yolov5m':
+            results = get_prediction_yolov5m(img_bytes)
+        results.save("static/")  # save the results
```

Comment:
We should avoid saving anything to the filesystem and instead of a redirection on static file we should return the raw bytes of data of the image with the correct `Content-Type` header. (`image/jpeg`)

https://developer.mozilla.org/fr/docs/Web/HTTP/Basics_of_HTTP/MIME_types